core/client/src/main/java/alluxio/client/block/LocalBlockInStream.java

```diff
@@ -62,10 +64,13 @@ public LocalBlockInStream(long blockId, long blockSize, WorkerNetAddress workerN
     try {
       mBlockWorkerClient = mCloser.register(mContext.createWorkerClient(workerNetAddress));
       LockBlockResult result = mBlockWorkerClient.lockBlock(blockId);
-      if (result == null) {
-        throw new IOException(ExceptionMessage.BLOCK_NOT_LOCALLY_AVAILABLE.getMessage(mBlockId));
-      }
       mReader = mCloser.register(new LocalFileBlockReader(result.getBlockPath()));
+    } catch (BlockDoesNotExistException e) {
+      mCloser.close();
+      throw new IOException(ExceptionMessage.BLOCK_NOT_LOCALLY_AVAILABLE.getMessage(mBlockId), e);
```

Comment:
Does this error message improve the readability? Since it is already a BlockDoesNotExistException?